docs/record-encryption-quickstart/index.adoc

```diff
@@ -0,0 +1,341 @@
+It's intended for developing and testing cloud & Serverless apps offline.
+
+[source,terminal,subs="attributes"]
+----
+$ helm repo add --force-update localstack-repo https://helm.localstack.cloud
+$ helm upgrade --install localstack localstack-repo/localstack --namespace kms --create-namespace --version {localstack-chart-version} --set service.type=ClusterIP --wait
+----
+
+NOTE: After installation, the LocalStack Helm chart prompts you to "Get the application URL". For this quickstart, you can ignore these steps.
+====
+
+.HashiCorp Vault
+[%collapsible]
+====
+
+HashiCorp Vault is available as a Cloud Service or standalone.  In this guide, we take install it standalone on Minikube.
+Note that Record Encryption requires the Transit Secrets Engine, which must be enabled.
+
+[source,terminal,subs="attributes"]
+----
+$ helm repo add --force-update hashicorp https://helm.releases.hashicorp.com
+$ helm upgrade --install vault hashicorp/vault --namespace kms --create-namespace --version {vault-chart-version} --set server.dev.enabled=true,server.dev.devRootToken=myroottoken --wait
+----
+
+And enable the Transit Secrets Engine.
+
+[source,terminal]
+----
+$ kubectl exec -n kms statefulsets/vault -- vault secrets enable transit
+----
+====
+
+=== Install the Strimzi Operator
+
+We are going to use https://strimzi.io/[Strimzi] to deploy the Kafka Cluster that will be proxied.
+Let's install Strimzi next:
+
+[source,terminal,subs="attributes"]
+----
+$ helm repo add --force-update strimzi https://strimzi.io/charts/
+$ helm upgrade --install strimzi-operator strimzi/strimzi-kafka-operator --namespace kafka --create-namespace --version {strimzi-version}
+----
+
+[#install_the_operator]
+=== Install the Kroxylicious Operator
+
+First, create a temporary directory for some working files.  We use a directory called `ko-install-nnnnn` beneath `/tmp`, but you can use any name and location you like.
+
+[source,terminal]
+----
+$ QUICKSTART_DIR=/tmp/ko-install-${RANDOM}
+$ mkdir ${QUICKSTART_DIR}
+$ cd ${QUICKSTART_DIR}
+----
+
+Now let's download and unpack the Kroxylicious Operator
+
+[source,terminal,subs="attributes"]
+----
+$ curl --fail --location --silent {OperatorAssetZipLink} --output kroxylicious-operator.zip
+$ unzip -q kroxylicious-operator.zip
+----
+
+And install the Kroxylicious Operator
+
+[source,terminal]
+----
+$ kubectl apply -f install
+$ kubectl wait --namespace kroxylicious-operator deployment/kroxylicious-operator --for=condition=Available=True --timeout=300s
+----
+
+[#deploy_the_cluster]
+== Deploy the Kafka Cluster
+
+Now we need a Kafka Cluster.
+This is the Kafka Cluster that will be proxied.
+We'll just use one from the Strimzi Quickstart.
+It will create a cluster in the `kafka` namespace.
+
+[source,terminal,subs="attributes"]
+----
+$ kubectl apply -n kafka -f https://raw.githubusercontent.com/strimzi/strimzi-kafka-operator/refs/tags/{strimzi-version}/examples/kafka/kafka-single-node.yaml
```

Comment:
question more than suggestion. We have defined a bunch of variables for github URLs for kroxylicious but have fully embedded it for the Strimzi download. Is there an advantage to either approach?

Reply:
My motivation for  using `{OperatorAssetZipLink}  was so that when automatically testing the quickstart against PR, substitute the Zip artefact resulting from the build (with a `file:` url) so that the quickstart will be tested against the to-be code.

Reply:
>  Is there an advantage to either approach?

I value the DRY type composition of the project github links.
```
:OperatorAssetZipLink: {github-release-downloads}/{OperatorAssetZipFileName}
```
It gives us a central place to update if we ever wanted to change the release asset locations.

External ones like the strimzi link feel like we can be a bit looser. For these ones I'd wait for multiple usages across guides before pulling them out into attributes.